<launch>
  <!-- simple_planning_simulator parameters -->
  <arg name="loop_rate" default="50.0"/>
  <arg name="simulation_frame_id" default="base_link"/>
  <arg name="map_frame_id" default="map"/>
  <arg name="initialize_source" default="RVIZ"/>
  <arg name="use_waypoints_for_z_position_source" default="true"/>
  <arg name="sim_steering_gear_ratio" default="15.0"/>

  <!-- model parameters -->
  <arg name="vehicle_model_type" default="DELAY_STEER_ACC"/>

  <arg name="tread_length" default="1.0"/>
  <arg name="angvel_lim" default="3.0"/>
  <arg name="vel_lim" default="50.0"/>
  <arg name="steer_lim" default="1.0"/>
  <arg name="angvel_rate" default="1.0"/>
  <arg name="accel_rate" default="7.0" />
  <arg name="steer_rate_lim" default="5.0"/>
  <arg name="vel_time_delay" default="0.25"/>
  <arg name="vel_time_constant" default="0.61"/>
  <arg name="acc_time_delay" default="0.3"/>
  <arg name="acc_time_constant" default="0.1"/>
  <arg name="steer_time_delay" default="0.24"/>
  <arg name="steer_time_constant" default="0.27"/>
  <arg name="angvel_time_delay" default="0.2"/>
  <arg name="angvel_time_constant" default="0.5"/>

  <arg name="add_measurement_noise" default="true"/>
  <arg name="pos_noise_stddev" default="0.01"/>
  <arg name="vel_noise_stddev" default="0.0"/>
  <arg name="rpy_noise_stddev" default="0.0001"/>
  <arg name="angvel_noise_stddev" default="0.00"/>
  <arg name="steer_noise_stddev" default="0.0001"/>


  <!-- simple_planning_simulator node -->
  <node pkg="simple_planning_simulator" type="simple_planning_simulator" name="simple_planning_simulator" output="screen">
    <remap from="/base_trajectory" to="/planning/scenario_planning/trajectory" />
    <remap from="~output/current_pose" to="current_pose" />
    <remap from="~output/current_twist" to="/vehicle/status/twist" />
    <remap from="~output/status" to="/vehicle/status" />
    <remap from="~input/vehicle_cmd" to="/control/vehicle_cmd" />

    <param name="loop_rate" value="$(arg loop_rate)"/>
    <param name="simulation_frame_id" value="$(arg simulation_frame_id)"/>
    <param name="map_frame_id" value="$(arg map_frame_id)"/>
    <param name="add_measurement_noise" value="$(arg add_measurement_noise)"/>
    <param name="initialize_source" value="$(arg initialize_source)"/>
    <param name="use_waypoints_for_z_position_source" value="$(arg use_waypoints_for_z_position_source)"/>
    <param name="sim_steering_gear_ratio" value="$(arg sim_steering_gear_ratio)"/>

    <param name="tread_length" value="$(arg tread_length)"/>
    <param name="angvel_lim" value="$(arg angvel_lim)"/>
    <param name="vel_lim" value="$(arg vel_lim)"/>
    <param name="steer_lim" value="$(arg steer_lim)"/>
    <param name="angvel_rate" value="$(arg angvel_rate)"/>
    <param name="steer_rate_lim" value="$(arg steer_rate_lim)"/>
    <param name="accel_rate" value="$(arg accel_rate)" />
    <param name="vel_time_delay" value="$(arg vel_time_delay)"/>
    <param name ="acc_time_delay" value="$(arg acc_time_delay)" />
    <param name="vel_time_constant" value="$(arg vel_time_constant)"/>
    <param name="acc_time_constant" value="$(arg acc_time_constant)"/>
    <param name="steer_time_delay" value="$(arg steer_time_delay)"/>
    <param name="steer_time_constant" value="$(arg steer_time_constant)"/>
    <param name="angvel_time_delay" value="$(arg angvel_time_delay)"/>
    <param name="angvel_time_constant" value="$(arg angvel_time_constant)"/>

    <param name="vehicle_model_type" value="$(arg vehicle_model_type)"/>
    <param name="pos_noise_stddev" value="$(arg pos_noise_stddev)"/>
    <param name="vel_noise_stddev" value="$(arg vel_noise_stddev)"/>
    <param name="rpy_noise_stddev" value="$(arg rpy_noise_stddev)"/>
    <param name="angvel_noise_stddev" value="$(arg angvel_noise_stddev)"/>
    <param name="steer_noise_stddev" value="$(arg steer_noise_stddev)"/>
  </node>

  <node pkg="topic_tools" type="relay" name="relay" args="/vehicle/status/twist /localization/twist" />
</launch>
